compiler/src/dotty/tools/dotc/typer/Checking.scala

```diff
@@ -799,7 +799,7 @@ trait Checking {
 
   /** Check that `tpt` does not define a higher-kinded type */
   def checkSimpleKinded(tpt: Tree)(implicit ctx: Context): Tree =
-    if (!tpt.tpe.hasSimpleKind && !ctx.compilationUnit.isJava) {
+    if (!tpt.tpe.dealias.hasSimpleKind && !ctx.compilationUnit.isJava) {
```

Comment:
Shouldn't `hasSimpleKind` be the one who does the dealiasing ?

Reply:
Good point.